.github/workflows/publish.yml

```diff
@@ -7,13 +7,19 @@ on:
       - dev-[0-9]+.[0-9]+.[0-9]+
     tags:
       - v*
+  pull_request:
+    types:
+      # publish images for PRs labeled "publish" whenever changed/labeled
+      [ opened, reopened, synchronize, labeled ]
   workflow_dispatch:
 
 env:
   REGISTRY: ghcr.io
 
 jobs:
   publish:
+    # for PRs, should only publish if it has "publish" label
+    if: ${{ (github.event_name != 'pull_request') || contains(github.event.pull_request.labels.*.name, 'publish') }}
```

Comment:
Nonblocking: in the Backend version, you made this check an envvar and output of the workflow

Reply:
Yeah - it was a necessary workaround there bc I wanted to reuse the variable. Not needed here since it's only used in one place.